tests/gtest/aviftest_helpers.cc

```diff
@@ -340,9 +341,15 @@ AvifImagePtr ReadImage(const char* folder_path, const char* file_name,
 }
 
 bool WriteImage(const avifImage* image, const char* file_path) {
-  return avifPNGWrite(file_path, image, /*requestedDepth=*/0,
-                      AVIF_CHROMA_UPSAMPLING_BEST_QUALITY,
-                      /*compressionLevel=*/0);
+  if (!image || !file_path) return false;
+  const size_t str_len = std::strlen(file_path);
+  if (str_len >= 4 && !std::strncmp(file_path + str_len - 4, ".png", 4)) {
```

Comment:
Nit: This comparison can be done with `memcmp` because both inputs are free of null characters in the first four bytes.